### ISSUE TITLE
Visualizations on W&B

### ISSUE THREAD

**Issue by AyushExel** [facebookresearch/detectron2#3404]
## 🚀 Feature
Allow visualization of training progress using media panels and tables. 

## Motivation & Examples
This is based on this [issue](https://github.com/facebookresearch/detectron2/issues/774). 
I'm an engineer at W&B and I've been working with object detection tasks. I regularly use some of the following visualizations on W&B dashboards. Would the following features be useful to other detection users? I'd like to know what the maintainers think.
### **Bounding box & Segmentation maps debugger**

W&B supports interactive media panels, where you can track how training progresses by adjusting the steps, confidence scores, and classes of predictions in real-time.
![ezgif com-gif-maker (2)](https://user-images.githubusercontent.com/15766192/130792944-77c59dbd-861b-4161-9c62-5f71e44bac7e.gif)

Try it live [here](https://wandb.ai/cayush/ert/runs/xc4m2xip?workspace=user-cayush)

The Media panel also supports debugging segmentation maps. 
![semantic segmentation (1)](https://user-images.githubusercontent.com/15766192/130792997-887b706c-13b2-4f18-8e7b-10ded88debfa.gif)

### Dataset visualization and versioning

With W&B tables, you can visualize, query, and filter your datasets in your browser.

Quickly compare results across different training epochs, datasets, hyperparameter choices, model architectures etc. For example, [take a look at this comparison of two models on the same test images →](https://wandb.ai/stacey/evalserver_answers_2/artifacts/results/eval_Daenerys/c2290abd3d7274f00ad8/files/eval_results.table.json#b6dae62d4f00d31eeebf$eval_Bob)

<img width="1142" alt="Screen Shot 2021-04-29 at 8 55 25 PM (1)" src="https://user-images.githubusercontent.com/15766192/130793194-eb6a77f3-11df-42a5-ae30-9f9da7b36bcf.png">

**Model versioning and DAGs**
![ezgif com-gif-maker (1) (1)](https://user-images.githubusercontent.com/15766192/130798266-0647a420-f5bd-442e-b556-d022f65fbdf3.gif)

We can have the user set 

- the model logging period
- and the desired metric they'd like to optimize.

Based on that, we'll log models after every  `model logging period` with alias `best` if the current model performs best on the `desired metric`

**Comment by AyushExel** [comment]
@ppwwyyxx what do you think about this? Would this be a useful add-on?

**Comment by nbardy** [comment]
Would be useful for me! I ended up adding it myself to debug some models.

**Comment by AyushExel** [comment]
@ppwwyyxx I've made a draft PR with a design proposal. Sorry for the dealy, was waiting on some UI changes
